### PR TITLE
Eng 2018 implement artifact repo

### PR DESCRIPTION
**Pull request by eunice-chan**
## Describe your changes and why you are making these changes
Implement sqlite + test cases
## Related issue number (if any)
Eng 2018

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [x] I have performed a self-review of my code.
- [N/A] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [x] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [N/A] All features on the UI continue to work correctly.
- [x] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

**Comment by eunice-chan** [comment]
It has the latest from main and https://github.com/aqueducthq/aqueduct/pull/722/ so it looks like I modified a lot of files if I set either as the base branch.

The files I've actually modified are:
- src/golang/lib/models/shared/artifact_type.go
- src/golang/lib/repos/artifact.go
- src/golang/lib/repos/sqlite/artifact.go
- src/golang/lib/repos/tests/random.go
- src/golang/lib/repos/tests/require.go
- src/golang/lib/repos/tests/seed.go
- src/golang/lib/repos/tests/suite.go

File: src/golang/lib/models/shared/artifact_type.go [hunk 1]
```diff
@@ -16,6 +16,20 @@ const (
 	PicklableArtifact ArtifactType = "picklable"
 )
 
+var ArtifactTypes = []ArtifactType{
+	UntypedArtifact,
```
Comment (saurav-c) [inline src/golang/lib/models/shared/artifact_type.go:20]: I feel like we should avoid defining this list if its just used for testing and instead define it in the testing package.

File: src/golang/lib/repos/sqlite/artifact.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+	return getArtifacts(ctx, DB, query, args...)
+}
+
+func (*artifactReader) GetByDAG(ctx context.Context, dagID uuid.UUID, DB database.Database) ([]models.Artifact, error) {
+	// Gets all artifacts that are a node with an incoming (id in `to_id`) or outgoing edge
+	// (id in `from_id`) in the `workflow_dag_edge` for the specified DAG.
+	query := fmt.Sprintf(
+		`SELECT %s FROM artifact WHERE id IN
+		(SELECT from_id FROM workflow_dag_edge WHERE workflow_dag_id = $1 AND type = '%s'
```
Comment (saurav-c) [inline src/golang/lib/repos/sqlite/artifact.go:64]: nit: can you indent everything in the subquery for readability

**Review by saurav-c — APPROVED** [review]
Left a couple nits otherwise LGTM

**Pull request merged**